### PR TITLE
Update sf units

### PR DESCRIPTION
**Pull request by rebeccafair**
Updating as a reuslt of https://github.com/pace-neutrons/Euphonic/pull/174, which changes the units of structure factors, so the test data needs to be updated. The tests will fail until the above PR is in Euphonic's master branch.

### CODE REVIEW

File: test/euphonic_horace_test.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def get_abspath(filename, sub_dir):
 bose = [None, False]
 negative_e = [None, True]
 conversion_mat = [None, (1./2)*np.array([[-1, 1, 1], [1, -1, 1], [1, 1, -1]])]
-lim = [None, 1e-9]
+lim = [None, 1e2]
```
Comment (cmarooney-stfc) [inline test/euphonic_horace_test.py]: Presuming this is a unit change, and although the general intent is given in the changelog, this is a raw number with no unit indications - may be worth commenting for this at end of line

File: test/euphonic_horace_test.py [hunk 2]
```diff
@@ -58,7 +58,14 @@ def get_expected_output_filename(material_name, pars, opts):
...
         fname += '_conv_mat_det' + str(np.linalg.det(opts['conversion_mat']))
     if opts['lim'] is not None:
-        fname += '_lim' + str(opts['lim'])
+        # Account for different str conversion of positive exponent
+        # sci notation - e.g. in Matlab string(1e2) = "100" but in
+        # Python str(1e2) = "100.0" unless explicitly cast to int
+        if opts['lim']%1 == 0:
+            strlim = str(int(opts['lim']))
+        else:
+            strlim = str(opts['lim'])
```
Comment (cmarooney-stfc) [inline test/euphonic_horace_test.py:77]: OK it has dawned on me that in python % works on reals and will give the decimal part of opts here. But for a naive C++ programmer this is very counter-intuitive. If you had a comment after "==0:" like # no fractional part, and after "else:" like # positive fractional part, that could help some maintenance programmer in the future.

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]
.mat files not reviewed. I would guess that they now give appropriate answers in the new units, but there are no tests here to allow checking this - it would be good to have such a test.

File: test/euphonic_horace_test.py (hunk 1 above)
Comment (rebeccafair) [reply test/euphonic_horace_test.py]: Done

File: test/euphonic_horace_test.py (hunk 2 above)
Comment (rebeccafair) [reply test/euphonic_horace_test.py:77]: Done

**Comment by rebeccafair** [comment]
I've added a test for a subset of parameters to assert that the new values are equal to the old values subject to a scale factor of `1e11/(2*n_atoms)` which is explained in the comments. This ended up with me refactoring a lot of the tests file to avoid duplicating code...

**Pull request merged**